traits/has_traits.py

```diff
@@ -416,14 +416,24 @@ def _property_method ( class_dict, name ):
 _HasTraits = None
 
 class MetaHasTraits ( type ):
-    ### JMS: Need a docstring here.
+    """ Controls the creation of HasTraits classes.
+
+    The heavy work is done by the `create_traits_meta_dict` function, which
+    takes the ``class_dict`` dictionary of class members and extracts and
+    processes the trait declarations in it. The trait declarations are then
+    added back to the class dictionary and passed off to the __new__ method
+    of the type superclass, to be added to the class.
+
+    """
     # All registered class creation listeners.
     #
     # { Str class_name : Callable listener }
     _listeners = {}
 
     def __new__ ( cls, class_name, bases, class_dict ):
-        mhto = MetaHasTraitsObject( cls, class_name, bases, class_dict, False )
+        traits_meta_dict = create_traits_meta_dict(
+            class_name, bases, class_dict, False )
+        class_dict.update(traits_meta_dict)
```

Comment:
These two lines are kinda weird: we're using `create_traits_meta_dict` both to mutate its input (the `class_dict`), *and* to return a new dict that we immediately merge with the `class_dict`. It would feel cleaner either to return a brand new dictionary that can be used immediately as our new class dict, or to mutate class_dict and return nothing.

I realise that we're hampered by the fact that this function is being used by the category machinery, too. I wonder whether it's worth having a clean version of this function that's just used by `MetaHasTraits`.